### ISSUE TITLE
64 bit Raspberry Pi 5

### ISSUE THREAD

**Issue by jankovalski** [mill1000/raspdif#26]
$ sudo raspdif

W: MAIN: 64 bit support is experimental. Please report any issues.
F: Memory: Failed to map physical address 0x10 of length 2080374784. Error: Invalid argument

$ uname -a
Linux raspberrypi 6.6.31+rpt-rpi-2712 #1 SMP PREEMPT Debian 1:6.6.31-1+rpt1 (2024-05-29) aarch64 GNU/Linux

$ cat /etc/debian_version 
12.6

Just compiled with make and no issue but it doesn't run. Any ideas?

**Comment by mill1000** [comment]
The Pi 5 SoC has a completely different architecture than previous generations.

Currently it's incompatible with raspdif and I have not looked into that it would take to add support.

**Comment by jankovalski** [comment]
Would it work if I switched to 32 bit OS or still no?

**Comment by mill1000** [comment]
No. This is a hardware compatibility problem, not an OS problem.
